scripts/bees

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+## Global vars
+export BEESHOME BEESSTATUS
+
+# Helpful functions
+ERRO(){ echo "ERROR:" "$@"; exit 1; }
+INFO(){ echo "INFO:" "$@"; }
+
+check_root_rights(){ [ "$UID" == "0" ] || ERRO "Must be runned as root!"; }
+bees_w(){
+    BEESHOME="$HOME" BEESSTATUS="$STATUS_FILE"
+    "$BIN_PATH" "$@"
+}
+
+## Default values
+HOME=/var/lib/bees/
+WORK_DIR=/run/bees/
+check_work_dir(){ [ ! -d "$WORK_DIR" ] && { mkdir -p "$WORK_DIR" || exit 1; }; }
+
+STATUS_FILE="$WORK_DIR/status"
+STATS_FILE="$HOME/beesstats.txt"
+check_stats(){ touch "$STATS_FILE"; chmod 644 "$STATS_FILE"; }
+
+MOUNT_DIR="$WORK_DIR/mounts/"
+DB_PATH="$HOME/beeshash.dat"
+DB_SIZE=1G
+
+check_db(){
+    touch "$DB_PATH"
+    chmod 700 "$DB_PATH"
+    # Check size
+    TMP="$(mktemp -u)"
+    CURRENT_SIZE="$(du -h --apparent-size $DB_PATH | sed 's/\t/ /g' | cut -d' ' -f1)"
+    NEW_SIZE="$DB_SIZE"
+    {
+        echo $NEW_SIZE
+        echo $CURRENT_SIZE
+    } | sort -h | tail -n 1 > "$TMP"
+    read -r BIGGER_SIZE < "$TMP"
+
+    [ "$BIGGER_SIZE" != "$NEW_SIZE" ] && ERRO "Can't shrink db file"
+    truncate -s "$NEW_SIZE" "$DB_PATH"
+}
+
+
+readonly BIN_PATH=/usr/lib/bees/bees
+readonly CONFIG=/etc/bees.conf
+
+# Array with FS UUIDs
+BTRFS_UUIDs=()
+
+## Prechecks
+[ ! -f "$BIN_PATH" ] && ERRO "Missing $BIN_PATH"
+source "$CONFIG" &> /dev/null || ERRO "Missing $CONFIG"
+
+## Check Args
+# if arg is a path, just try run bees
+if mountpoint -q "$1"; then
+    check_root_rights
+    check_work_dir
+    check_db
+    check_stats
+    bees_w "$@"
+    exit 0
+fi
+
+case "$1" in
+    start) : ;;
+    status)
+        [ ! -f "$STATUS_FILE" ] && ERRO "Can't find bees status, bees running?"
+        cat "$STATUS_FILE"
+        exit 0
+    ;;
+    stats)
+        [ ! -f "$STATS_FILE" ] && ERRO "Can't find bees stats, bees running?"
+        cat "$STATS_FILE"
+        exit 0
+    ;;
+    *)
+        echo "bees (<path1> <path2>| arg)"
+        echo "      <path>  - use /etc/bees.conf but run bees for <path>"
+        echo "      start   - run with /etc/bees.conf"
+        echo "      status  - current status of bees"
+        echo "      stats   - current db stats"
+        exit 0
+    ;;
+esac
+
+if (( ${#BTRFS_UUIDs[@]} == 0 )); then
+    ERRO "BTRFS_UUIDs are empty, nothing to do"
+fi
+
+BASE="/sys/fs/btrfs/"
+
+for UUID in "${BTRFS_UUIDs[@]}"; do
+    [ ! -d "$BASE/$UUID" ] && ERRO "Can't find BTRFS UUID: $UUID"
+done
+
+check_root_rights
+check_work_dir
+check_db
+check_stats
+
+trap force_umount_all SIGINT SIGTERM EXIT
+force_umount_all(){
+    for path in "$MOUNT_DIR"/*; do
+        mountpoint -q "$path" && umount -l $path
+    done
+}
+
+## Make mounts
+INFO "Mounts base dir: $MOUNT_DIR"
+BASE="/dev/disk/by-uuid/"
+for UUID in "${BTRFS_UUIDs[@]}"; do
+    [ ! -b "$BASE/$UUID" ] && ERRO "Missing: $BASE/$UUID"
+    mkdir -p "$MOUNT_DIR/$UUID"
+    mount -osubvolid=5 "$BASE/$UUID" "$MOUNT_DIR/$UUID" || exit 1
+done
+
+cd "$MOUNT_DIR"
+bees_w ./*-*-*-*-*
```

Comment:
Then, i prefer to have a subvol inside a root of FS with custom name.
I will implement the code with several configs, i.e. config per FS UUID
Thanks.